### PR TITLE
DHP-1063

### PR DESCRIPTION
**Pull request by hallieswan**
- add workflow to run tests
- increase timeout for slow tests
- bump caniuse-lite version
- clean up Time component

### CODE REVIEW

File: .github/workflows/build-and-test.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          cache: yarn
+      - name: build
+        run: |
+          export CI=false
+          yarn install --check-files
+          yarn build
```
Comment (hallieswan) [inline .github/workflows/build-and-test.yaml:25]: Match setup in build-and-deploy workflow

File: .github/workflows/build-and-test.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      - name: test
+        run: |
+          yarn craco test --watchAll=false --silent
```
Comment (hallieswan) [inline .github/workflows/build-and-test.yaml:28]: Don't run tests in watch mode and hide console warnings/errors for ease of reading test results.

File: src/components/adherence/summary/AdherenceAlerts.test.tsx [hunk 3]
```diff
@@ -157,7 +157,7 @@ describe('AdherenceAlerts', () => {
         'UNREAD',
         loggedInSessionData.token
       )
-    })
+    }, 10_000)
```
Comment (hallieswan) [inline src/components/adherence/summary/AdherenceAlerts.test.tsx:160]: Increase timeout for slow running tests

File: src/components/studies/scheduler/SchedulableSingleSessionContainer.test.tsx [hunk 4]
```diff
@@ -27,6 +27,9 @@ const studySession: StudySession = {
       title: 'Word Meaning',
       tags: ['vocabulary'],
       minutesToComplete: 2,
+      labels: [],
+      isLocal: false,
+      isReadOnly: false,
```
Comment (hallieswan) [inline src/components/studies/scheduler/SchedulableSingleSessionContainer.test.tsx:32]: Add fields that were recently added to `Assessment` type

File: src/components/surveys/survey-design/question-edit/rhs-subcontrols/Time.tsx [hunk 5]
```diff
@@ -57,12 +58,11 @@ const ValueSelector: React.FunctionComponent<{
 
       <StyledDropDown
         labelId={CONFIG[type].labelId}
-        value={value}
+        value={value || ''}
```
Comment (hallieswan) [inline src/components/surveys/survey-design/question-edit/rhs-subcontrols/Time.tsx:61]: When `value` is `undefined`, MUI warns that an 'out-of-range value' has been specified for the select component and recommends passing an empty string in this case.

File: src/components/surveys/survey-design/question-edit/rhs-subcontrols/Time.tsx [hunk 6]
```diff
@@ -129,7 +129,11 @@ const Time: React.FunctionComponent<{
     if (val) {
       setRange(undefined)
       setError('')
-      //onUpdateFormat(undefined)
+      onUpdateFormat({
+        ...step.inputItem.formatOptions,
+        maximumValue: undefined,
+        minimumValue: undefined,
+      })
```
Comment (hallieswan) [inline src/components/surveys/survey-design/question-edit/rhs-subcontrols/Time.tsx:136]: After checking "No min or max validation", update the step to reflect that there is no `minimumValue` or `maximumValue` set.

image

File: yarn.lock [hunk 7]
```diff
@@ -3719,9 +3719,9 @@ caniuse-api@^3.0.0:
     lodash.uniq "^4.5.0"
 
 caniuse-lite@^1.0.0, caniuse-lite@^1.0.30001286, caniuse-lite@^1.0.30001297:
-  version "1.0.30001400"
-  resolved "https://registry.npmjs.org/caniuse-lite/-/caniuse-lite-1.0.30001400.tgz"
-  integrity sha512-Mv659Hn65Z4LgZdJ7ge5JTVbE3rqbJaaXgW5LEI9/tOaXclfIZ8DW7D7FCWWWmWiiPS7AC48S8kf3DApSxQdgA==
+  version "1.0.30001540"
+  resolved "https://registry.npmjs.org/caniuse-lite/-/caniuse-lite-1.0.30001540.tgz"
+  integrity sha512-9JL38jscuTJBTcuETxm8QLsFr/F6v0CYYTEU6r5+qSM98P2Q0Hmu0eG1dTG5GBUmywU3UlcVOUSIJYY47rdFSw==
```
Comment (hallieswan) [inline yarn.lock:3724]: Update [caniuse-lite](https://www.npmjs.com/package/caniuse-lite), so browsers database is up to date for checking compatibility across different browsers.

**Review by syoung-smallwisdom — APPROVED** [review]
Looks good

File: .github/workflows/build-and-test.yaml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          cache: yarn
+      - name: build
+        run: |
+          export CI=false
```
Comment (nickgros) [inline .github/workflows/build-and-test.yaml:23]: Why CI=false?

**Review by nickgros — APPROVED** [review]
LGTM

File: .github/workflows/build-and-test.yaml (hunk 8 above)
Comment (hallieswan) [reply .github/workflows/build-and-test.yaml:23]: When `CI=true`, warnings are treated as errors. The build produces many warnings, so the app will only compile when `CI=false`.  I opened [a ticket](https://sagebionetworks.jira.com/browse/DHP-1113) to clean up the warnings, so we can set `CI=true`.

**Pull request merged**